### PR TITLE
improve jindoruntime networkmode and hostPID usage

### PR DESCRIPTION
**Pull request by frankleaf**
Signed-off-by: frankleaf <frank.wt@alibaba-inc.com>

### Ⅴ. Special notes for reviews

### CODE REVIEW

File: api/v1alpha1/jindoruntime_types.go [hunk 1]
```diff
@@ -170,6 +170,10 @@ type JindoRuntimeSpec struct {
 	// +kubebuilder:validation:Enum=HostNetwork;"";ContainerNetwork
 	// +optional
 	NetworkMode NetworkMode `json:"networkmode,omitempty"`
+
+	// Whether to enable hostPID or not
+	// +optional
+	DisableHostPID bool `json:"disableHostPID,omitempty"`
```
Comment (cheyang) [inline api/v1alpha1/jindoruntime_types.go:176]: Do you want to expose it to the enduser? May I know the reason?

File: pkg/ddc/jindo/transform.go [hunk 2]
```diff
@@ -553,9 +554,19 @@ func (e *JindoEngine) transformToken(runtime *datav1alpha1.JindoRuntime, value *
...
-	expectedPortNum := 2
 
+	// if not usehostnetwork then use default port
+	if !value.UseHostNetwork {
+		value.Master.Port.Rpc = DEFAULT_MASTER_RPC_PORT
+		value.Worker.Port.Rpc = DEFAULT_WORKER_RPC_PORT
+		if value.Master.ReplicaCount == JINDO_HA_MASTERNUM {
+			value.Master.Port.Raft = DEFAULT_RAFT_RPC_PORT
+		}
+		return nil
```
Comment (cheyang) [inline pkg/ddc/jindo/transform.go:565]: Did you test it? Can the port be released successfully?

File: api/v1alpha1/jindoruntime_types.go (hunk 1 above)
Comment (frankleaf) [reply api/v1alpha1/jindoruntime_types.go:176]: In some environment, it doesn't support hostPID

File: pkg/ddc/jindo/transform.go (hunk 2 above)
Comment (frankleaf) [reply pkg/ddc/jindo/transform.go:565]: the ContainerNetwork port can be released automatically? I am not sure

Comment (cheyang) [reply pkg/ddc/jindo/transform.go:565]: Could you test it? thanks.FYI @TrafalgarZZZ

File: api/v1alpha1/jindoruntime_types.go (hunk 1 above)
Comment (cheyang) [reply api/v1alpha1/jindoruntime_types.go:176]: I suggest to name it `hostPID` which is the same with Pod API.

File: pkg/ddc/jindo/transform.go [hunk 3]
```diff
@@ -95,6 +95,7 @@ func (e *JindoEngine) transform(runtime *datav1alpha1.JindoRuntime) (value *Jind
 		},
 		Owner: transfromer.GenerateOwnerReferenceFromObject(runtime),
 	}
+	e.transformNetworkMode(runtime, value)
```
Comment (cheyang) [inline pkg/ddc/jindo/transform.go:98]: Please cover this logic with unit test.

**Review by cheyang — APPROVED** [review]

**Pull request merged**